src/main/kotlin/org/rust/ide/inspections/RsTraitImplementationInspection.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections
+
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.PsiElementVisitor
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.RsImplItem
+import org.rust.lang.core.psi.RsVisitor
+import org.rust.lang.core.psi.ext.TraitImplementationInfo
+import org.rust.lang.core.psi.ext.resolveToTrait
+import org.rust.lang.core.psi.ext.selfParameter
+import org.rust.lang.core.psi.ext.valueParameters
+import org.rust.lang.utils.RsDiagnostic
+import org.rust.lang.utils.Severity
+import org.rust.lang.utils.addToHolder
+
+class RsTraitImplementationInspection : RsLocalInspectionTool() {
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor = object : RsVisitor() {
+        override fun visitImplItem(impl: RsImplItem) {
+            val traitRef = impl.traitRef ?: return
+            val trait = traitRef.resolveToTrait ?: return
+            val traitName = trait.name ?: return
+
+            val implInfo = TraitImplementationInfo.create(trait, impl) ?: return
+
+            if (implInfo.missingImplementations.isNotEmpty()) {
+                val missing = implInfo.missingImplementations.mapNotNull { it.name?.let { "`$it`" } }.joinToString(", ")
+                val endElement = impl.typeReference ?: impl.impl
+                RsDiagnostic.TraitItemsMissingImplError(impl.impl, endElement, missing, impl, Severity.ERROR)
+                    .addToHolder(holder)
+                endElement.nextSibling?.also {
+                    RsDiagnostic.TraitItemsMissingImplError(it, impl.lastChild, missing, impl, Severity.INFO)
```

Comment:
Interesting thing: if you change severity of this inspection via settings, this annotation won't change it